### PR TITLE
[BEAM-121] DisplayData API tweaks

### PR DESCRIPTION
**Pull request by swegner**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace "<Jira issue #>" in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by swegner** [comment]
R: @bjchambers

**Comment by bjchambers** [comment]
I see test failures. Can you verify that nothing is broken?

File: sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/runners/DataflowPipelineTranslator.java [hunk 1]
```diff
@@ -730,12 +730,8 @@ private void addOutput(String name, PValue value, Coder<?> valueCoder) {
     }
 
     private void addDisplayData(String name, DisplayData displayData) {
-      List<Map<String, Object>> serializedItems = Lists.newArrayList();
-      for (DisplayData.Item item : displayData.items()) {
-        serializedItems.add(MAPPER.convertValue(item, Map.class));
-      }
-
-      addList(getProperties(), name, serializedItems);
+      List list = MAPPER.convertValue(displayData, List.class);
```
Comment (bjchambers) [inline sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/runners/DataflowPipelineTranslator.java]: List<?>

File: sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/runners/DataflowPipelineTranslator.java [hunk 2]
```diff
@@ -730,12 +730,8 @@ private void addOutput(String name, PValue value, Coder<?> valueCoder) {
...
+      addList(getProperties(), name, list);
```
Comment (bjchambers) [inline sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/runners/DataflowPipelineTranslator.java]: Looks like the build error is in this file too:
```
slave/workspace/beam_PreCommit_MavenVerify/sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/runners/DataflowPipelineTranslator.java:84:8: Unused import: com.google.common.collect.Lists.
```

**Comment by swegner** [comment]
Ok, this should build clean now. @bjchambers, can you take another look? Once this is in I'll rebase other PR's on top and we can have another round of feedback on those.

File: sdks/java/core/src/test/java/com/google/cloud/dataflow/sdk/transforms/display/DisplayDataTest.java [hunk 3]
```diff
@@ -743,21 +824,90 @@ public void populateDisplayData(Builder builder) {
...
+            .add("class", DisplayDataTest.class)
+              .withLinkUrl("http://abc")
+              .withLabel("baz")
+        ;
+      }
+    };
+    DisplayData data = DisplayData.from(component);
+
+    JsonNode json = MAPPER.readTree(MAPPER.writeValueAsBytes(data));
+    assertThat(json, hasItem(expectedJson(component.getClass(), "STRING", "string", "\"foobar\"")));
```
Comment (bjchambers) [inline sdks/java/core/src/test/java/com/google/cloud/dataflow/sdk/transforms/display/DisplayDataTest.java]: This test may be easier to read (and verify it matches expectations) if it just has the expected value as a single string, or a string for each value.

**Comment by swegner** [comment]
I've addressed all feedback so far. Please take another look. @bjchambers

**Comment by bjchambers** [comment]
LGTM

**Comment by davorbonaci** [comment]
@swegner, did we backport this? (Cc: @bjchambers)

**Comment by swegner** [comment]
I haven't done any backporting work for recent PRs-- I'm not quite familiar with the process. I'll work with @bjchambers on this.

**Comment by swegner** [comment]
Backported via https://github.com/GoogleCloudPlatform/DataflowJavaSDK/pull/216

**Pull request closed without merging**